orangecontrib/spectroscopy/widgets/owhyper.py

```diff
@@ -784,14 +893,13 @@ def extract_col(data, var):
         # d = image_values(data).X[:, 0]
         parts = []
         for slice in split_to_size(len(data), 10000):
-            part = image_values(data[slice]).X[:, 0]
+            part = image_values(data[slice]).X
             parts.append(part)
             progress_interrupt(0)
-        d = np.concatenate(parts)
+            d = np.concatenate(parts)
```

Comment:
Why was the `np.concatenate` moved into the loop? This only makes things slower, as it copies array contents multiple times instead of only once.

Reply:
Good catch.